controller/orphan_controller.go

```diff
@@ -398,46 +399,52 @@ func (oc *OrphanController) cleanupOrphanedReplicaInstance(orphan *longhorn.Orph
 		}
 	}()
 
-	instance, imName, err := oc.extractOrphanedInstanceInfo(orphan)
+	instanceName, instanceUUID, imName, err := oc.extractOrphanedInstanceInfo(orphan)
 	if err != nil {
-		return false, err
+		oc.logger.WithError(err).Warnf("Illegal replica orphan %v, deleting the CR without cleaning up the instance", orphan.Name)
+		return true, nil
 	}
 
 	var status *longhorn.InstanceStatus
-	if replicaCR, err := oc.ds.GetReplicaRO(instance); err != nil {
+	if replicaCR, err := oc.ds.GetReplicaRO(instanceName); err != nil {
 		if !datastore.ErrorIsNotFound(err) {
 			return false, err
 		}
 		status = nil
 	} else {
 		status = &replicaCR.Status.InstanceStatus
 	}
-	oc.cleanupOrphanedInstance(orphan, instance, imName, longhorn.InstanceManagerTypeReplica, status)
+	oc.cleanupOrphanedInstance(orphan, instanceName, instanceUUID, imName, longhorn.InstanceManagerTypeReplica, status)
 	return true, nil
 }
 
-func (oc *OrphanController) extractOrphanedInstanceInfo(orphan *longhorn.Orphan) (name, instanceManager string, err error) {
+func (oc *OrphanController) extractOrphanedInstanceInfo(orphan *longhorn.Orphan) (name, uuid, instanceManager string, err error) {
 	name, ok := orphan.Spec.Parameters[longhorn.OrphanInstanceName]
 	if !ok {
-		return "", "", fmt.Errorf("failed to get instance name for instance orphan %v", orphan.Name)
+		return "", "", "", fmt.Errorf("failed to get instance name for instance orphan %v", orphan.Name)
 	}
 
 	instanceManager, ok = orphan.Spec.Parameters[longhorn.OrphanInstanceManager]
 	if !ok {
-		return "", "", fmt.Errorf("failed to get instance manager for instance orphan %v", orphan.Name)
+		return "", "", "", fmt.Errorf("failed to get instance manager for instance orphan %v", orphan.Name)
 	}
 
 	switch orphan.Spec.DataEngine {
 	case longhorn.DataEngineTypeV1, longhorn.DataEngineTypeV2:
 		// supported data engine type
 	default:
-		return "", "", fmt.Errorf("unknown data engine type %v for instance orphan %v", orphan.Spec.DataEngine, orphan.Name)
+		return "", "", "", fmt.Errorf("unknown data engine type %v for instance orphan %v", orphan.Spec.DataEngine, orphan.Name)
+	}
```

Comment:
Seems redundant, as it should have already been validated by the webhook.

Reply:
Could we assume all the content form a CR is valid? Indeed there's no chance to modify the orphan CR spec in our current design.

Reply:
> Could we assume all the content form a CR is valid? Indeed there's no chance to modify the orphan CR spec in our current design.

Could we assume `orphan.Spec.DataEngine` is valid?

Reply:
Removing it should be fine.
Can you add a validating logic in the webhook to make the field immutable?

Reply:
Immutable would be great. Let me include this.